app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,14 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(_('Invalid provided user type of %s ') % options[:user].class.to_s)
+           end
+    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless user.mail_enabled?
```

Comment:
In this case, I'd rather log a warning to logs, this will break mailing rake process executed from job rendering mailer unable to continue with the task if the exception is not catched. Logging is safer.

Reply:
done